### ISSUE TITLE
Latest ruff 0.0.266 fails to import

### ISSUE THREAD

**Issue by jaraco** [charliermarsh/ruff#4394]
I was working on jaraco/jaraco.packaging#11 when suddenly the tests started failing with `ModuleNotFoundError`. I traced it to the recent 0.0.266 release:

```
 ~ $ pip-run -v ruff -- -c "import ruff"
Collecting ruff
  Using cached ruff-0.0.266-py3-none-macosx_10_9_x86_64.macosx_11_0_arm64.macosx_10_9_universal2.whl (10.1 MB)
Installing collected packages: ruff
Successfully installed ruff-0.0.266
Traceback (most recent call last):
  File "<string>", line 1, in <module>
ModuleNotFoundError: No module named 'ruff'
```

**Comment by madkinsz** [comment]
Confirmed this reproduces on `0.0.266` with `python -c "import ruff"` but not on `0.0.265`.

**Comment by jaraco** [comment]
I can see the binary there, but no library:

```
 ~ $ pip-run ruff
Python 3.11.3 (main, Apr  7 2023, 20:13:31) [Clang 14.0.0 (clang-1400.0.29.202)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import sys
>>> sys.path
['', '/var/folders/sx/n5gkrgfx6zd91ymxr2sr9wvw00n8zm/T/pip-run-h27rhwxt', '/opt/homebrew/Cellar/python@3.11/3.11.3/Frameworks/Python.framework/Versions/3.11/lib/python311.zip', '/opt/homebrew/Cellar/python@3.11/3.11.3/Frameworks/Python.framework/Versions/3.11/lib/python3.11', '/opt/homebrew/Cellar/python@3.11/3.11.3/Frameworks/Python.framework/Versions/3.11/lib/python3.11/lib-dynload', '/opt/homebrew/Cellar/python@3.11/3.11.3/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages']
>>> import pathlib
>>> libs = pathlib.Path(sys.path[1])
>>> list(libs.iterdir())
[PosixPath('/var/folders/sx/n5gkrgfx6zd91ymxr2sr9wvw00n8zm/T/pip-run-h27rhwxt/bin'), PosixPath('/var/folders/sx/n5gkrgfx6zd91ymxr2sr9wvw00n8zm/T/pip-run-h27rhwxt/__pycache__'), PosixPath('/var/folders/sx/n5gkrgfx6zd91ymxr2sr9wvw00n8zm/T/pip-run-h27rhwxt/sitecustomize.py'), PosixPath('/var/folders/sx/n5gkrgfx6zd91ymxr2sr9wvw00n8zm/T/pip-run-h27rhwxt/ruff-0.0.266.dist-info')]
>>> list(libs.joinpath('bin').iterdir())
[PosixPath('/var/folders/sx/n5gkrgfx6zd91ymxr2sr9wvw00n8zm/T/pip-run-h27rhwxt/bin/ruff')]
```

**Comment by jaraco** [comment]
Can I expect a rapid mitigation (fix or yank) or should I invest in working around the issue?

**Comment by madkinsz** [comment]
@charliermarsh looks like the `site-packages/ruff` directory is missing.

@jaraco why do you use `import ruff`?

**Comment by edgarrmondragon** [comment]
fwiw this is causing problems for [`nbqa-ruff`](https://github.com/nbQA-dev/nbQA) too

**Comment by charliermarsh** [comment]
Alright, I'll go ahead and revert while we figure this out.

`python -m ruff file.py` is also failing.

**Comment by charliermarsh** [comment]
Yanked for now.

**Comment by charliermarsh** [comment]
I'll have a new release out within the next hour.

**Comment by jaraco** [comment]
> Yanked for now.

Super helpful and unblocked my work. Thanks for the super-rapid response.

**Comment by charliermarsh** [comment]
Thanks for raising the issue so quickly, and apologies for the churn.

I did do some manual testing of the wheels prior to cutting the release, since we upgraded Maturin, but I didn't test the use of Ruff-as-module, and it turns out we didn't have any automated testing for that either, which @madkinsz has already improved :)

**Comment by charliermarsh** [comment]
Fixed in v0.0.267.

**Comment by jaraco** [comment]
> @jaraco why do you use `import ruff`?

I use [pytest-ruff](https://pypi.org/project/pytest-ruff), which [imports ruff](https://github.com/buserbrasil/pytest-ruff/blob/895a6dd2164e9d020cdf5f8c717b6216de6f825b/pytest_ruff.py#L4).
